app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java

```diff
@@ -509,11 +510,26 @@ private void handleUpdatedFeedStatus(Feed feed) {
                 viewBinding.subscribeButton.setEnabled(true);
                 viewBinding.subscribeButton.setText(R.string.open_podcast);
                 if (didPressSubscribe) {
+                    didPressSubscribe = false;
+                    if (UserPreferences.isEnableAutodownload()) {
+                        Feed feed1 = DBReader.getFeed(getFeedId(feed));
+                        FeedPreferences feedPreferences = feed1.getPreferences();
+                        feedPreferences.setAutoDownload(viewBinding.autoDownloadCheckBox.isChecked());
+                        feed1.savePreferences();
+                    }
                     openFeed();
                 }
             } else {
                 viewBinding.subscribeButton.setEnabled(true);
                 viewBinding.subscribeButton.setText(R.string.subscribe_label);
+                if (UserPreferences.isEnableAutodownload()) {
+                    LinearLayout.LayoutParams layoutParams
+                            = (LinearLayout.LayoutParams) viewBinding.subscribeButton.getLayoutParams();
+                    layoutParams.setMargins(16, 16, 16, 0);
```

Comment:
This still modifies the margins from Java code, which is not really elegant (and does not work correctly, depending on screen resolution). Could you please try to find a way to do this from XML?